### ISSUE TITLE
New proto3 keyword: stream

### ISSUE THREAD

**Issue by JakeWharton** [square/wire#282]
https://github.com/grpc/grpc-java/blob/master/examples/src/main/proto/route_guide.proto#L51

I don't see this in the formal proto3 docs anywhere.

**Comment by JakeWharton** [comment]
Also the `{` and `}` are new.

**Comment by EddieRingle** [comment]
The `stream` keyword correlates to the following in MethodDescriptorProto:

```protobuf
  // Identifies if client streams multiple client messages
  optional bool client_streaming = 5 [default=false];
  // Identifies if server streams multiple server messages
  optional bool server_streaming = 6 [default=false];
```

**Comment by JakeWharton** [comment]
Any idea why it isn't documented? New as of proto3-alpha-3.1 maybe?

**Comment by EddieRingle** [comment]
Not sure. Relevant compiler function is here: https://github.com/google/protobuf/blob/master/src/google/protobuf/compiler/parser.cc#L1725

Blame says the `stream`-relevant bits were added as a result of syncing up with Google's internal repo. Might be a feature they don't want to announce yet?

**Comment by JakeWharton** [comment]
This is now in the proto2 and (incomplete in the) proto3 grammar. There's both the stream keyword on the rcp types as well as a stream keyword inside of a service.

**Comment by JakeWharton** [comment]
Fixed by #580.
